app/services/migrate_legacy_token_service.rb

```diff
@@ -0,0 +1,9 @@
+class MigrateLegacyTokenService
+  def migrate_legacy_tokens(tokens)
```

Comment:
```suggestion
  attr_reader :tokens
  
  def initialize(tokens)
    @tokens = tokens
  end
  
  def perform
```

ça me parait mieux en terme d'interface.

Reply:
Je voulais le rendre stateless vu que c'est un service.

Reply:
`MigrateLegacyTokenService.new.migrate_legacy_tokens(tokens)` vs `MigrateLegacyTokenService.new(tokens).perform` c'est la même modulo plus dry/consistant avec le reste de la basecode en terme d'interface.

Btw, je ne vois pas la différence ici niveau stateless :D (si t'avais fait un truc du genre `MigrateLegacyTokenService.migrate_legacy_tokens(tokens)` ok

Reply:
J'ai clairement pas l'habitude de passer des objets dans les `initialize` voir même je trouve ca bizarre. Qu'on y passe d'autres classe pour faire de la composition je comprends mais des variables, j'ai l'habitude d'évité même si j'ai remarqué que tu le faisais systématiquement.

Là dessus je me base surtout sur le bouquin DDD qui définit les services comme ca, après par conformité je peux changer pour être iso. Faudra juste que je fasse un effort pour les prochains d'y penser :D

Reply:
Mais en vrai osef par conformité j'adapte, faudra que je m'efforce de faire comme ca c'est tout :D

Reply:
Honnêtement je m'en fiche, juste ton interface niveau wording est bizarre, tu peux faire un truc du genre `MigrateLegacyTokenService.new.perform(tokens)` ou `(MigrateLegacyTokenService.call(tokens)` ou whatever d'autre c'est ~ish pareil et plus lisible